src/importlinter/adapters/user_options.py

```diff
@@ -29,7 +36,10 @@ def read_options(self, config_filename: Optional[str] = None) -> Optional[UserOp
         for config_filename in config_filenames:
             config = configparser.ConfigParser()
             file_contents = settings.FILE_SYSTEM.read(config_filename)
-            config.read_string(file_contents)
+            try:
+                config.read_string(file_contents)
+            except configparser.Error:
```

Comment:
Why is this `try except` needed?

Reply:
So, the UserOptionReader are executed one by one. If one returns None, then the next is tried. The configparser cannot parse TOML and will therefore fail with an exception

Reply:
I'm confused by the need for this too. This would only be checking the `potential_config_filenames`, which would be in INI format?

Reply:
The reason why I added this is the following: We don't have a chance to know if a file is a TOML or ini file. The configparser will fail if it's parsing a TOML file. 
Therefore each reader will try to parse the file and will not fail but just return None if the file structure is not supported. If no reader can read the file, import-linter will print a message.

If this code is not in place and the user specifies a TOML file over the CLI, the code will fail with an internal exception

Reply:
> We don't have a chance to know if a file is a TOML or ini file.

I don't think this is true? Both `setup.cfg` and `pyproject.toml` have known formats, no? That only leaves `.importlinter`, which I think for the time being we can just say needs to be in INI format. Then the `potential_config_filenames` lists the filenames expected in that format.

Reply:
I got your point. There is one additional exception: If the user provides a file, we can guess but not know which file format it is. We could add a kind of pre-evaluation logic but that doesn't feel right